### DISCUSSION TITLE
Interpolating between two datasets

### DISCUSSION THREAD

**Discussion by stackjohn** [Unidata/siphon#746] [Q&A] [answered]
Hello, 

Is it possible to generate intermediate datasets via interpolation?

For example lets say we have two datasets 1 hour apart, using something like this 

```
from datetime import datetime, timedelta

# Create a new NCSS query
query = ncss.query()

# Request data in netCDF format
query.accept('netcdf')

# Ask for our variable
query.variables('Temperature_isobaric')

# Ask for the 500 hPa surface
query.vertical_level(50000)

# Set the time range of data we want
now = datetime.utcnow()
query.time_range(now, now + timedelta(hours=1))

# Set the spatial limits
query.lonlat_box(west=-110, east=-45, north=50, south=10)

# get the data!
data = ncss.get_data(query)
```
From the example [here](https://unidata.github.io/python-training/workshop/Model_Output/downloading-model-fields-with-ncss/)

Can we generate interpolated datasets for each minute between now and one hour plus?

**Comment by dopplershift** [comment, marked as answer]
I would take `data` and open it with xarray and use `Dataset.resample` to do the interpolation:
```python
from xarray.backends import NetCDF4DataStore
import xarray as xr

ds = xr.open_dataset(NetCDF4DataStore(data))
ds.resample(time='1min')
```
I'm not an xarray expert, but [based on the docs](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.resample.html), that looks like it should work, provided the dimension is called "time".

**Reply by stackjohn** [reply]
Great, thank you. I will give it a try 👍
